src/Test.UnitTests.Sarif/Visitors/SortingVisitorTests.cs

```diff
@@ -0,0 +1,306 @@
+                result.RuleIndex.Should().Be(ruleIndex);
+            }
+
+            // make sure artifactLocation index points to right artifacts
+            IList<Artifact> artifacts = sortedLog1.Runs.First().Artifacts;
+            foreach (Result result in sortedLog1.Runs.First().Results)
+            {
+                ArtifactLocation artifactLoc = result?.Locations?.First()?.PhysicalLocation?.ArtifactLocation;
+                if (artifactLoc != null && artifactLoc.Index != -1)
+                {
+                    artifactLoc.Uri.Should().Be(artifacts[artifactLoc.Index].Location.Uri);
+                }
+            }
+        }
+
+        [Fact]
+        public void SortingVisitor_NullEmptyListTests()
+        {
+            // arrange
+            // create a log with all results have same value.
+            SarifLog sarifLog = new SarifLog
+            {
+                Runs = new[]
+                {
+                    new Run
+                    {
+                        Results = new[]
+                        {
+                            new Result
+                            {
+                                RuleId = "TESTRULE001"
+                            },
+                            new Result
+                            {
+                                RuleId = "TESTRULE001"
+                            },
+                            new Result
+                            {
+                                RuleId = "TESTRULE001"
+                            },
+                        },
+                    },
+                },
+            };
+
+            // setup results
+            IList<Result> results = sarifLog.Runs[0].Results;
+            results[0].Locations = new[] { new Location { Message = new Message { Text = "test location" } } };
+            results[1].Locations = null;
+            results[2].Locations = new List<Location>();
+
+            // act
+            SarifLog sortedLog = new SortingVisitor().VisitSarifLog(sarifLog);
+
+            // assert
+            // if collection elements all values are same expect a child list
+            // the order should depend on list values.
+            // expected order: null < empty < collection has element
+            results = sortedLog.Runs[0].Results;
+            results[0].Locations.Should().BeNull();
+            results[1].Locations.Should().BeEmpty();
+            results[2].Locations.Should().NotBeEmpty();
+        }
+
+        private static SarifLog CreateTestSarifLog(Random random)
+        {
+            SarifLog sarifLog = RandomSarifLogGenerator.GenerateSarifLogWithRuns(random, runCount: 1, resultCount: random.Next(100));
+
+            CreateCodeFlows(sarifLog.Runs?.First()?.Results, sarifLog.Runs?.First()?.Artifacts, random);
+
+            return sarifLog;
+        }
+
+        private static void CreateCodeFlows(IList<Result> results, IList<Artifact> artifacts, Random random)
```

Comment:
>CreateCodeFlows [](http://example.com/codeflow?start=28&length=15)

Find an elegant way to merge this capability into the RandomSarifLogFileGenerator.

Reply:
Add default enum parameter to RandomSarifLogGenerator it generate additional test data based on the enum value.